### DISCUSSION TITLE
日本語音声合成での韻律記号の扱いについて

### DISCUSSION THREAD

**Discussion by RRR-troisR** [espnet/espnet#4594] [Q&A] [answered]
ESPNetのTTSモデルは例えば以下のように
        >>> pyopenjtalk_g2p_prosody("こんにちは。")
        ['^', 'k', 'o', '[', 'N', 'n', 'i', 'ch', 'i', 'w', 'a', '$']
音素と韻律記号の両方を１つの形にまとめて扱っていますが、韻律記号は継続長を持たないので、例えばMonotonic Alignment Searchなど継続長を推論をする場面において悪影響が出るように思うのですが、それらはどのような形で対応をしているのでしょうか？

**Comment by kan-bayashi** [comment, marked as answer]
ご質問ありがとうございます！
現状は特に特別な扱いはしていませんので、韻律記号にも継続長が割り当てられます。
おもしろいことに、こういった韻律記号を入れてもAttention alignmentやMonotonic Alignmentは普通に動きます。
FastSpeech系で強制アライメントの結果ではなく、ARモデルのAttentionから継続長を計算しているのもこの辺の処理も気にせずに行えるというところが大きいです。
が、もちろん階層的なシンボル表現にして足し合わせて使うというような案もあると思います。
(この辺の扱いの違いを私自身は調査したことがあまりありません)
